### ISSUE TITLE
Dialog with text input scrolls to bottom after closing with ESC in Safari

### ISSUE THREAD

**Issue by roydigerhund** [tailwindlabs/headlessui#2506]
**What package within Headless UI are you using?**
@headlessui/react

**What version of that package are you using?**
v1.7.14

**What browser are you using?**
Safari Version 16.4 (18615.1.26.11.23)

**Reproduction URL**
https://github.com/roydigerhund/dialog-safari-scroll-bug

Can be tested on Tailwind UI as well
https://tailwindui.com

**Describe your issue**

When a Dialog with an text input field is opened and then closed via ESC without writing something into the input, the page will scroll all the way to the bottom.

**Comment by thecrypticace** [comment]
Hey! Thanks for reporting this. It's a super weird one and I spent a good portion of yesterday and today looking into this. Unfortunately, this is a Safari bug. It can occur Vite, Next, or even plain HTML/JS projects. The presence of content below the fold under certain conditions is the trigger for this.

There isn't really any way for us to work around this as far as I know. All the solutions I found involve hiding the elements that cause the problem using `display: none` or `visibility: hidden` but those can end up being be actual content.

I've [filled a bug](https://bugs.webkit.org/show_bug.cgi?id=257217) with Webkit and hopefully they'll pick it up soon and look into it.

Sorry I don't have better news for you but I'll try to keep an eye on it and, if it's soon enough and I remember to, I'll report back if/when they do eventually fix it.
